### PR TITLE
unsampled report downloads #44

### PR DESCRIPTION
**Pull request by j450h1**
See discussion on unsampled report downloads #44

### CODE REVIEW

**Comment by j450h1** [comment]
Updated to use reportTitle as main argument instead of documentid

File: R/unsampled.R [hunk 1]
```diff
@@ -54,115 +54,127 @@ ga_unsampled_list <- function(accountId,
...
-    html <- GET(document[["content"]][["webContentLink"]],
-                add_headers(Authorization=document[["request"]][["headers"]][["Authorization"]]))
-    #Read and parse html for confirmation code
-    stop_for_status(html)
-    too_large_html <- content(html, "text")
-    pat <- "&amp;confirm=(.*?)&amp;"
-    confirm_code <- regmatches(too_large_html, gregexpr(pat, too_large_html))
-    pat <- "&amp;"
-    confirm_code <- gsub(pat, "", confirm_code[[1]])
+  too_large <- (as.numeric(document[["content"]][["fileSize"]]) / 1048576) >= 25 # bytes to MB  if(too_large){
```
Comment (MarkEdmondson1234) [inline R/unsampled.R]: @j450h1 is this if block not needed anymore?  the too_large check?

**Comment by MarkEdmondson1234** [comment]
I enabled Drive API for the default googleAnalyticsR project for this

File: tests/testthat/test_unsampled_report_downloads.R [hunk 2]
```diff
@@ -8,33 +11,65 @@
...
+#   accountId,
+#   webPropertyId,
+#   profileId,
+#   downloadFile = FALSE
+# )
+# 
+# # Save to list of dataframes using pipes
+# dfs <- ga_unsampled_list(accountId, webPropertyId, profileId) %>%
+#   select(title) %>%
+#   unlist() %>%
```
Comment (MarkEdmondson1234) [inline tests/testthat/test_unsampled_report_downloads.R]: We can avoid this unlist by using `dplyr::pull`

Comment (MarkEdmondson1234) [reply tests/testthat/test_unsampled_report_downloads.R]: e.g. 

```
ga_unsampled_list(accountId = "54019251", 
                          webPropertyId = "UA-54019251-4", 
                          profileId = "129374148") %>%
    dplyr::pull(title) %>%
    purrr::map(ga_unsampled_download, 
        accountId = "54019251", 
        webPropertyId = "UA-54019251-4", 
        profileId = "129374148", 
        downloadFile = FALSE)
```

**Comment by MarkEdmondson1234** [comment]
I adapted the tests that you can see in https://github.com/MarkEdmondson1234/googleAnalyticsR/commit/781e23381504d3227f1261bd6fbe801a2dd313ad

File: R/unsampled.R (hunk 1 above)
Comment (j450h1) [reply R/unsampled.R]: No it is still required as this additional warning page pops up that needs to be parsed to get the final download link. I think i just adjusted the formatting/spacing here.

File: tests/testthat/test_unsampled_report_downloads.R (hunk 2 above)
Comment (j450h1) [reply tests/testthat/test_unsampled_report_downloads.R]: Great, i forgot about pull.

**Comment by j450h1** [comment]
Thanks for adapting tests.

File: R/unsampled.R (hunk 1 above)
Comment (MarkEdmondson1234) [reply R/unsampled.R]: Ok this is probably broke at the moment then, I haven't a 25MB file to test it yet

**Pull request merged**